### PR TITLE
Cherry pick cert file config from master to release-1.1

### PR DESCRIPTION
**Pull request by drichelson**
Cherry pick from master: Configuration: no longer hardcode mesh certs original PR: #12189

### CODE REVIEW

**Comment by drichelson** [comment]
cc @rshriram since you approved the original PR

**Comment by drichelson** [comment]
@rshriram Any idea how to get "merges-blocked-needs-admin Expected — Waiting for status to be reported Required" resolved? it looks like that is preventing this from getting merged.

**Comment by drichelson** [comment]
@ZackButcher @diemtvu @ymesika It looks like all required checks are passing. Any reason not to merge this?

**Comment by drichelson** [comment]
@rshriram I now see a new Required check that is not reporting its status: 
`prow/e2e-bookInfoTests-trustdomain.sh Expected — Waiting for status to be` 
Any help is appreciated

**Comment by pnambiarsf** [comment]
@rshriram - This PR is blocking us from proceeding with bare metal services support.  Would appreciate your help in unblocking us. Thanks.

**Comment by pnambiarsf** [comment]
@diemtvu , @ymesika  and @ZackButcher - Are you able to unblock us ?

**Comment by pnambiarsf** [comment]
@linsun - is anyone able to help unblock us ?

**Comment by louiscryan** [comment]
@wattli @myidpt can one of you help unblock this please

**Comment by drichelson** [comment]
@wattli thanks for the 👍 However it looks like we are awaiting 2 checks. These are the same 2 checks we were awaiting for many days.

**Comment by wattli** [comment]
@mandarjog should know more about the pending test issue.

**Comment by pnambiarsf** [comment]
@duderino @rshriram - Could one of you merge this ? All tests have passed.

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -478,6 +540,37 @@ func init() {
...
+		if now.After(endWait) {
+			log.Warna("certificates still not available after", maxWait)
+			break
+		}
+		if now.After(nextLog) {
+			log.Infof("waiting for certificates")
+			logDelay = logDelay * 2
+			nextLog.Add(logDelay)
+		}
+		time.Sleep(100 * time.Millisecond)
```
Comment (duderino) [inline pilot/cmd/pilot-agent/main.go]: Do you want to sleep in seconds?

**Pull request merged**